server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2702,7 +2702,7 @@ public void snapshotRemoteStoreIndexShard(
                         indexTotalNumberOfFiles,
                         indexTotalFileSize,
                         store.indexSettings().getUUID(),
-                        store.indexSettings().getRemoteStoreRepository(),
+                        store.indexSettings().getRemoteSegmentStoreDataRepository(),
```

Comment:
i think we need to add another parameter for metadata repository as well since lock files are in metadata repo. currently the `remoteStoreRepository` is used for both source data and lock files.

https://github.com/linuxpi/OpenSearch/blob/08e8325f03cd7f0c205c74bd9892677fb3cd4b59/server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java#L669-L675

Reply:
Right! Thanks for catching this, let me make the required change.

Reply:
Lock files are stored separately stored and not in metadata repo .

Reply:
Lock file path is generated with segment repository when data and metadata is the same repo. 
Once we separate data and metadata, we need to associate lock files with one of these repos. As per the comment: https://github.com/opensearch-project/OpenSearch/pull/12993#discussion_r1555237010, it makes more sense to have lock files stored in metadata repo.